buckaroo/Solver.fs

```diff
@@ -23,14 +24,49 @@ module Solver =
     Hints : AsyncSeq<Atom * PackageLock>;
   }
 
-  type SearchStrategy = ISourceExplorer -> SolverState -> AsyncSeq<PackageIdentifier * (PackageLocation * Set<Version>)>
+  type SearchStrategyError =
+  | NotSatisfiable of NotSatisfiable
+  | NoLocationAvaiable of PackageIdentifier * Constraint
```

Comment:
What is the difference between these two cases?

Reply:
both are errors that are emitted if we couldn't find a location for PackageIdentifer + Constraint.
However we need to distinguish between those two as one may never be satisfiable the other might if a new location is discovered. A new Location might be discovered if we pick a different version of some package.

We need to write proper tests for this case as it may require that we establish a  certain ordering between PackageIdentifers

Reply:
OK, makes sense.